sklearn/cluster/hierarchical.py

```diff
@@ -704,9 +729,21 @@ def fit(self, X, y=None):
                              " instance, got 'memory={!r}' instead.".format(
                                  type(memory)))
 
-        if self.n_clusters <= 0:
+        # if distance threshold is set we set n_clusters to None
```

Comment:
This was to pass the check "if both n_cluster and distance_threshold are set then fail". n_clusters_ does not need to be set to None for any computation as it gets ignored.